api/konnect/v1alpha1/konnect_cloudgateway_transitgateway_types.go

```diff
@@ -0,0 +1,199 @@
+type KonnectTransitGatewayAPISpec struct {
+	// Type is the type of the Konnect transit gateway.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=awsTransitGateway;azureTransitGateway
+	Type TransitGatewayType `json:"type"`
+
+	// AWSTransitGateway is the configuration of an AWS transit gateway.
+	// Used when type is "AWS Transit Gateway".
+	//
+	// +kubebuilder:validation:Optional
+	AWSTransitGateway *AWSTransitGateway `json:"awsTransitGateway,omitempty"`
+	// AzureTransitGateway is the configuration of an Azure transit gateway.
+	// Used when type is "Azure Transit Gateway".
+	//
+	// +kubebuilder:validation:Optional
+	AzureTransitGateway *AzureTransitGateway `json:"azureTransitGateway,omitempty"`
+}
+
+// AWSTransitGateway is the configuration of an AWS transit gateway.
+type AWSTransitGateway struct {
+	// Human-readable name of the transit gateway.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=120
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// List of mappings from remote DNS server IP address sets to proxied internal domains, for a transit gateway
+	// attachment.
+	//
+	// +kubebuilder:validation:Optional
+	DNSConfig []TransitGatewayDNSConfig `json:"dns_config,omitempty"`
+	// CIDR blocks for constructing a route table for the transit gateway, when attaching to the owning
+	// network.
+	//
+	// +kubebuilder:validation:Required
+	CIDRBlocks []string `json:"cidr_blocks"`
+	// configuration to attach to AWS transit gateway on the AWS side.
+	//
+	// +kubebuilder:validation:Required
+	AttachmentConfig AwsTransitGatewayAttachmentConfig `json:"attachment_config"`
+}
+
+// AzureTransitGateway is the configuration of an Azure transit gateway.
+type AzureTransitGateway struct {
+	// Human-readable name of the transit gateway.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=120
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// List of mappings from remote DNS server IP address sets to proxied internal domains, for a transit gateway
+	// attachment.
+	//
+	// +kubebuilder:validation:Optional
+	DNSConfig []TransitGatewayDNSConfig `json:"dns_config,omitempty"`
+	// configuration to attach to Azure VNET peering gateway.
+	//
+	// +kubebuilder:validation:Required
+	AttachmentConfig AzureVNETPeeringAttachmentConfig `json:"attachment_config"`
+}
+
+// TransitGatewayDNSConfig is the DNS configuration of a tansit gateway.
+type TransitGatewayDNSConfig struct {
+	// Remote DNS Server IP Addresses to connect to for resolving internal DNS via a transit gateway.
+	RemoteDNSServerIPAddresses []string `json:"remote_dns_server_ip_addresses,omitempty"`
+	// Internal domain names to proxy for DNS resolution from the listed remote DNS server IP addresses,
+	// for a transit gateway.
+	DomainProxyList []string `json:"domain_proxy_list,omitempty"`
```

Comment:
If a field is meant to be optional, let's mark it explicitly as such. (ditto for all the other fields)

```suggestion
	// +kubebuilder:validation:Optional
	DomainProxyList []string `json:"domain_proxy_list,omitempty"`
```

Reply:
Done.